app/src/main/java/com/codelab/android/datastore/TasksApplication.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.codelab.android.datastore
+
+import android.app.Application
+import androidx.datastore.preferences.SharedPreferencesMigration
+import androidx.datastore.preferences.preferencesDataStore
+
+private const val USER_PREFERENCES_NAME = "user_preferences"
+
+class TasksApplication : Application() {
+    val dataStore by preferencesDataStore(
```

Comment:
What's the reasoning for declaring this in the Application class? Could it make sense to instead declare it at the top level in the UserPreferencesRepository class?